boa/src/builtins/number/mod.rs

```diff
@@ -555,6 +555,221 @@ impl Number {
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-number.isfinite
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite
+    pub(crate) fn number_is_finite(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(Value::Boolean(if let Some(val) = args.get(0) {
+            match val {
+                Value::Integer(_) => true,
+                Value::Rational(number) => number.is_finite(),
+                Value::Object(ref obj) => {
+                    let obj = obj.borrow();
+                    if let Some(number) = obj.as_number() {
+                        number.is_finite()
+                    } else {
+                        false
+                    }
+                }
+                _ => false,
+            }
+        } else {
+            false
+        }))
+    }
+
+    /// `Number.isInteger( number )`
+    ///
+    /// Checks if the argument is an integer.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-number.isinteger
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger
+    pub(crate) fn number_is_integer(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(Value::Boolean(if let Some(val) = args.get(0) {
+            Number::is_integer(val).is_some()
+        } else {
+            false
+        }))
+    }
+
+    /// `Number.isNaN( number )`
+    ///
+    /// Checks if the argument is a number, returning false if it isn't.
+    ///
+    /// If the number is NaN true is returned.
+    ///
+    /// Otherwise false is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isnan-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN
+    pub(crate) fn number_is_nan(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(Value::Boolean(if let Some(val) = args.get(0) {
+            match val {
+                Value::Integer(_) => false,
+                Value::Rational(number) => number.is_nan(),
+                Value::Object(ref obj) => {
+                    let obj = obj.borrow();
+                    if let Some(number) = obj.as_number() {
+                        number.is_nan()
+                    } else {
+                        false
+                    }
+                }
+                _ => false,
+            }
+        } else {
+            false
+        }))
+    }
```

Comment:
```suggestion
    pub(crate) fn number_is_nan(
        _this: &Value,
        args: &[Value],
        _ctx: &mut Interpreter,
    ) -> ResultValue {
        if let Some(Value::Boolean(number)) = args.get(0) {
            return Ok(number.is_nan().into());
        }
        
        Ok(Value::from(false))
    }
```
same here `Number.isNaN()` does not check for number object only primitive number https://tc39.es/ecma262/#sec-isnan-number